chore(docs): create architecture page (#28481)
apache/superset: docs/docs/installation/architecture.mdx

---
title: Architecture
hide_title: true
sidebar_position: 1
version: 1
---

import useBaseUrl from "@docusaurus/useBaseUrl";

# Architecture

This page is meant to give new administrators an understanding of Superset's components.

## Components

A Superset installation is made up of these components:

> 1. The Superset application itself

Reviewer: Should we call it "the Superset [python] backend"?

Reviewer: Should we call out "The Superset frontend" - a collection of React application bundled with webpack and using antd, ...

Reviewer 2: I mention this in a suggestion below in the description of the Superset application.

> 1. The Superset application itself

Reviewer 2: ```suggestion
1. **The Superset application itself**. This includes the Python (Flask) backend application, API layer, and the React frontend, built via Webpack, and static assets needed for the application to work.
```

> 2. A metadata database to store Superset's data about users, charts, dashboards, etc.

Reviewer 2: ```suggestion
2. **A metadata database**. This stores Superset's data about users, charts, dashboards, etc.
```

> 3. A caching layer (optional, but necessary for some features)

Reviewer: this is called out twice in "components" and "optional component", i think it's a great idea to seperate the section, but let's have each component on just one side

> 3. A caching layer (optional, but necessary for some features)

Reviewer 2: ```suggestion
3. **A caching layer**. Optional, but necessary for some features. There is more information about this below.
```

4. A worker & beat (optional, but necessary for some features)

### Optional components and associated features

The optional components above are necessary to enable these features:
- [Alerts and Reports](/docs/configuration/alerts-reports)
- [Caching](/docs/configuration/cache)
- [Async Queries](/docs/configuration/async-queries-celery/)
- [Dashboard Thumbnails](/docs/configuration/cache/#caching-thumbnails)

If you install with Kubernetes or Docker Compose, all of these components will be created.

However, installing from PyPI only creates the application itself. Users installing from PyPI will need to configure a caching layer, worker, and beat on their own if they wish to enable the above features. Configuration of those components for a PyPI install is not currently covered in this documentation.

Here are further details on each component.

### The Superset Application

This is the core application. Superset operates like this:
- A user visits a chart or dashboard
- That triggers a SQL query to the data warehouse holding the underlying dataset

> - The resulting data is served up in a data visualization

Reviewer 2: ```suggestion
- The resulting data is served up in a data visualization
- The Superset application is comprised of the Python (Flask) backend application (server), API layer, and the React frontend, built via Webpack, and static assets needed for the application to work
```
